internal/cachecontroller/cache_controller.go

```diff
@@ -100,15 +103,29 @@ func (c *InMemoryCacheController) DetermineInvalidation(
 		return entry.LastModified
 	}
 
-	lastModified, err, _ := c.sf.Do(storeID, func() (interface{}, error) {
-		return c.findChangesAndInvalidate(ctx, storeID)
-	})
-
-	if err != nil {
-		return time.Time{}
+	c.mu.Lock()
+	_, present := c.inflightInvalidations[storeID]
+	if !present {
+		c.inflightInvalidations[storeID] = struct{}{}
 	}
+	c.mu.Unlock()
+	if !present {
+		span.SetAttributes(attribute.Bool("checkInvalidation", true))
+		// if the cache cannot be found, we want to invalidate entries in the background
+		// so that it does not block the answer path.
+		go func() {
+			// important to propagate the cancel without cancel to avoid
+			// cancelling the invalidation when parent request is complete.
+			_ = c.findChangesAndInvalidate(context.WithoutCancel(ctx), storeID)
```

Comment:
This is the only usage of `findChangesAndInvalidate` and we are not using the `err` returned, why not update the function's signature?

Alternatively, could you leave a comment explaining why it's okay to ignore the error?